pkg/v1/remote/options.go

```diff
@@ -193,3 +194,11 @@ func WithProgress(updates chan<- v1.Update) Option {
 		return nil
 	}
 }
+
+// WithNoTagsLimit disables request tags limit when listing image tags.
+//
+// The default behavior is a limit of 1000 tags.
+func WithNoTagsLimit(o *options) error {
```

Comment:
Probably want something like "WithPageSize", where anything <= 0 indicates no size. Then we can have 1000 as the default page size.

Reply:
Yes, I agree; your approach covers more cases.
I have reworked the logic and added support for the catalog call.